### ISSUE TITLE
Spatial Transformer example - locnet activation functions

### ISSUE THREAD

**Issue by 9thDimension** [EderSantana/seya#46]
See `In [5]` of https://github.com/EderSantana/seya/blob/master/examples/Spatial%20Transformer%20Networks.ipynb where the localisation network is defined.

Is there a reason why the `Convolution2D` layers have no activations? And the final layer (responsible for regressing the affine transformation parameters) _does_ have a `'relu'` activation. I may be wrong, but I thought that it's typical for the final layer of a neural net regression to have linear activation.

I asked some others about this, but nobody could explain why the activations are laid out this way, and they suggested I raise it here -- so hopefully the author can comment on these design choices.

**Comment by EderSantana** [comment]
STNs are really unstable. The gradients we get there are just not well behaved. I think that avoiding as much nonlinearities as possible was just the way the original authors of the paper found to get it to behave well. 

So yeah, STNs are mostly not well understood yet. Maybe STNs are harder to train than GANs. The research community should be paying more attention to that soon. But now, when designing your STNs I'd recommend using the hyperparameters in the original paper and its follow ups.

**Comment by 9thDimension** [comment]
Fair enough. I have found in other unrelated experiments that nets with linear activations can be perform useful tasks and are fast to optimize.

So in that case, why do you have the `'relu'` activation before the locnet's final regression layer?
I can't find where they suggest such an idea in the original paper.
